tests/commands/test_patch.py

```diff
@@ -0,0 +1,63 @@
+"""
+:Description: Tests the `patch` CLI
+"""
+
+import json
+
+import pytest
+from click.testing import CliRunner
+
+from conda_recipe_manager.commands.patch import patch
+from tests.file_loading import TEST_FILES_PATH
+from tests.smoke_testing import assert_cli_usage
+
+
+def test_usage() -> None:
+    """
+    Smoke test that ensures rendering of the help menu
+    """
+    assert_cli_usage(patch)
+
+
+def test_faulty_recipe_file(fs: pytest.Function) -> None:
+    """
+    Test to check the patch operation fails with a fauly recipe i.e a recipe which does not have the correct format etc
+    """
+    runner = CliRunner()
+    fs.add_real_directory(TEST_FILES_PATH, read_only=False)  # type: ignore[attr-defined]
+
+    json_patch_path = TEST_FILES_PATH / "patch" / "correct_json_patch.json"
+    faulty_recipe_file_path = TEST_FILES_PATH / "patch" / "incorrect_recipe.yaml"
+
+    runner.invoke(patch, [str(json_patch_path), str(faulty_recipe_file_path)])
+    assert pytest.raises(IOError)
```

Comment:
Shouldn't this be a `with pytest.raises()`? Does this achieve the same thing? I actually don't know off the top of my head.

Reply:
Actually the more I think about it, this `assert` shouldn't be here at all. We shouldn't be throwing in a CLI, we should be returning a POSIX-style error code.